libs/platform/i18n/src/lib/locale/default-locale.service.ts

```diff
@@ -57,35 +88,24 @@ export class DefaultLocaleService implements LocaleService {
     this.setActive$.next(value);
   }
 
-  formatDate(stamp: string | number): Observable<string> {
-    return this.get().pipe(
-      map((locale) =>
-        Intl.DateTimeFormat(locale.replace('_', '-')).format(new Date(stamp))
-      )
+  formatDate(stamp: string | number | Date): Observable<string> {
+    return this.dateFormat$.pipe(
+      map((dateFormat) => dateFormat.format(new Date(stamp)))
```

Comment:
Seems like missing check to avoid extra date conversion:
```suggestion
      map((dateFormat) => stamp instanceof Date ? stamp : new Date(stamp))
```